h2o-algos/src/main/java/hex/pca/PCA.java

```diff
@@ -314,18 +309,23 @@ public void computeImpl() {
           // Compute SVD of Gram A'A/n using JAMA library
           // Note: Singular values ordered in weakly descending order by algorithm
           _job.update(1, "Calculating SVD of Gram matrix locally");
-          Matrix gramJ = _wideDataset ? new Matrix(ogtsk._gram.getXX()) : new Matrix(gtsk._gram.getXX());
-          SingularValueDecomposition svdJ = gramJ.svd();
+          DenseMatrix gramJ = _wideDataset ? new DenseMatrix(ogtsk._gram.getXX()) : new DenseMatrix(gtsk._gram.getXX());
```

Comment:
For fat and wide datasets, it is faster to calculate the SVD of AA'/n instead of A'A/n.  However, the algorithm will decide on its own without user input.  That is why the _wideDataset variable is there.  Currently, I have not considered any sparse/dense matrix tradeoff.